composer/callbacks/checkpoint_saver.py

```diff
@@ -510,17 +577,23 @@ def _save_checkpoint(self, state: State, logger: Logger):
                 ).lstrip('/') + '.symlink'
 
                 # create and upload a symlink file
-                with tempfile.TemporaryDirectory() as tmpdir:
-                    symlink_filename = os.path.join(tmpdir, 'latest.symlink')
-                    # Sharded checkpoints for torch >2.0 use directories not files for load_paths
-                    if state.fsdp_sharded_state_dict_enabled:
-                        src_path = str(pathlib.Path(remote_file_name).parent)
+                symlink_filename = os.path.join(
+                    self.tmp_dir_for_symlink.name,
+                    f'latest.{self.count}.symlink',
+                )
+                # Sharded checkpoints for torch >2.0 use directories not files for load_paths
+                if state.fsdp_sharded_state_dict_enabled:
+                    src_path = str(pathlib.Path(remote_file_name).parent)
+                else:
+                    src_path = remote_file_name
+                log.debug(f'Creating symlink file {symlink_filename} -> {src_path}')
+                this_rank_saves_symlinks = dist.get_global_rank() == 0 or not state.fsdp_sharded_state_dict_enabled
```

Comment:
```suggestion
                rank_saves_symlinks = dist.get_global_rank() == 0 or not state.fsdp_sharded_state_dict_enabled
```
nit conciseness. Also why is this different from the class attribute? why does it need to be rechecked?

Reply:
because we don't know `self.rank_saves_symlinks` is true or not in the __init__. We only know once it runs into this code. I can add a check if `self.rank_saves_symlinks` is set or not, but i don't think it's very necessary.